lib/LoginStrategy.ts

```diff
@@ -53,13 +53,13 @@ export interface LoginStrategyOptions extends StrategyOptions {
   acceptedRoles?: string[];
   includeAllRoles?: boolean;
   jwtSecret?: string | Buffer;
-  jwtSignOptions?: jwt.SignOptions;
+  jwtSignOptions?: SignOptions;
 }
 
 export class LoginStrategy extends BaseStrategy {
   private readonly claimField: string;
   private readonly jwtSecret?: string | Buffer;
-  private readonly jwtSignOptions?: jwt.SignOptions;
+  private readonly jwtSignOptions?: SignOptions;
```

Comment:
We should perhaps set `jwtSignOptions` to a default of `RS256` if undefined as otherwise this update is a breaking change

Reply:
@nichonien to elaborate here, I think this is update is a breaking change if the consumer of `passport-did-auth` hasn't previously set `RS256` in their SignOptions. It seems that `jsonwebtoken` now requires it to be set (or the default was changed to `HS256`), so if clients haven't set it to `RS256` then their code would break if they just updated the `passport-did-auth` version and did nothing else.
Does that make sense from your perspective or do you see it differently?

Reply:
> We should perhaps set jwtSignOptions to a default of RS256 if undefined as otherwise this update is a breaking change 

@jrhender It is upto the implementor of PDA (in our case `did-auth-proxy`) to decide on the which signing `algo` to use, PDA doesn't have any requirement which makes it necessary to have a default `algo`. Also `jsonwentoken` has a default set to use `HS256`, so we are just passing the default back to it (if we set any). We can definitely do a good documentation here. IMO it is not a breaking change.

> It seems that jsonwebtoken now requires it to be set (or the default was changed to HS256)

The `HS256` has been default for very long - https://github.com/auth0/node-jsonwebtoken/blob/88bc965061ed65299a395f42a100fb8f8c3c683e/sign.js#L88. To me it is still unclear why our test started failing after this bump (I might have missed something). 

> so if clients haven't set it to RS256 then their code would break if they just updated the passport-did-auth version and did nothing else

`did-auth-proxy` uses `HS256` - https://github.com/energywebfoundation/did-auth-proxy/blob/b33bde1505de00f5223e3bfc7123a3eab9e13723/authorization-server/src/modules/auth/strategies/auth.strategy.ts#L43 . 

IMO, improving the documentation would be the right thing to do.

Reply:
Ok, thanks for clarifying that `HS256` has been the default for some time. This is a strange situation but assuming that the default was `HS256` before and hasn't been changed, then you're right that this wouldn't be a break change. I think we can proceed as is then 👍